spec/features/update_username_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'Update username controller', type: :feature, js: true do
+  let!(:user) { create(:user) }
+
+  before do
+    login_as user
+  end
+
+  context 'Update Username page' do
+    it 'checks if page is visited' do
+      visit '/update_username'
+      expect(page).to have_content('Enter Username to update')
+    end
+
+    it 'visits and enter updated username' do
+      VCR.use_cassette('update_username') do
+        create(:user, id: 500, username: 'Ragesoss')
+        new_username = 'New Username'
+        User.find(500).update_attribute(:username, new_username)
+        visit '/update_username'
+        fill_in('username', with: new_username)
+        click_button 'Update Username'
+        expect(User.find(500).username).to eq(new_username)
```

Comment:
This test doesn't make sense. The form should update the username to whatever their current username is according to Wikimedia servers, based on their global_id. But in this example, the user does not have a global_id and the test looks for the same username ("New Username") that user 500 already had before using the form.

Reply:
So What i'll do is i'll create a new user and assign some global_id to it as well then I'll update its username using update_attribute. then I'll check. Will that work?

Reply:
You can use the example in user_importer_spec.rb line 149 as an example.

Reply:
@ragesoss I've updated it in that way which you mentioned